common/path/platform/c++17/path.cpp

```diff
@@ -42,14 +42,25 @@ bool IsAbsolute(const string &path) {
 	return fs::path(path).is_absolute();
 }
 
-expected::ExpectedUnorderedSet<string> ListFiles(
+expected::ExpectedBool IsExecutable(const string &file_path) {
+	fs::perms perms = fs::status(file_path).permissions();
+	if ((perms & (fs::perms::owner_exec | fs::perms::group_exec | fs::perms::others_exec))
+		== fs::perms::none) {
+		log::Warning("'" + file_path + "' is not executable");
+		return false;
+	}
+	return true;
+}
+
+expected::ExpectedStringVector ListFiles(
```

Comment:
It is a set, though, isn't it?

Reply:
I guess it's because we need to sort it?

Reply:
Yes, it was my initial impl.

I changed it to save a conversion from `set` to `vector` before sorting it.

But I agree, if this is to be general it should be `unordered_set`. 

One other opportunity could be templating the return type, but maybe over-complicating it.

Reply:
I like `unordered_set` here because it reflects what the directory content actually is -- no duplicate entries and no guaranteed order.

Reply:
Agreed actually. I'll change it.